### DISCUSSION TITLE
Validation rules not working

### DISCUSSION THREAD

**Discussion by amigolj** [Laravel-Backpack/community-forum#790] [Q&A (Help)]
Hi could you please help with this rule `required_if` is not working
![image](https://github.com/Laravel-Backpack/community-forum/assets/10392966/d204e958-beb0-4669-97ae-0e0d39751b1f)

**Comment by jcastroa87** [comment]
Hello @amigolj 

First, what are you trying to do exactly? Can you share your migration, model, and a return from `php artisan backpack:version` and a little explanation about what you are expecting to happen here?

Cheers.

**Comment by amigolj** [comment]
@jorgetwgroup sorry I'm the first time to use this feature of github.
I need if `section_menu_id` greater than `0` set field `meal_menu_id`  like required. So when the user fills `section_menu_id` greater than 0, he should fill `meal_menu_id`, but if he fills  `section_menu_id` equal 0, in this case not required fill `meal_menu_id`

![image](https://github.com/Laravel-Backpack/community-forum/assets/10392966/e05ebcff-cfee-40dd-b480-69881339efb3)

It's model:
`<?php

namespace App\Models;

use Backpack\CRUD\app\Models\Traits\CrudTrait;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Support\Facades\Log;
use Illuminate\Support\Str;

const ESTIMATE_MENU_ITEM = 'estimate_menu_item';

class ItemMenu extends Model
{
    use CrudTrait;
    use HasFactory;

    /*
    |--------------------------------------------------------------------------
    | GLOBAL VARIABLES
    |--------------------------------------------------------------------------
    */

    protected $table = 'item_menu';
    // protected $primaryKey = 'id';
    // public $timestamps = false;
    protected $guarded = ['id'];
    // protected $fillable = [];
    // protected $hidden = [];
    public static function boot()
    {

        parent::boot();

        static::creating(function ($item) {
            Log::alert("Creating menu",[$item]);
            $this->setFields($item);
        });

        static::updating(function ($item) {
            //
        });

    }
    /*
    |--------------------------------------------------------------------------
    | FUNCTIONS
    |--------------------------------------------------------------------------
    */
    public function getAllByMealId($mealId):array
    {
        return $this->where('meal_menu_id', $mealId)->get()->toArray();
    }
    /*
    |--------------------------------------------------------------------------
    | RELATIONS
    |--------------------------------------------------------------------------
    */
    public function sectionMenu() {
        return $this->belongsTo(SectionMenu::class, 'section_menu_id');
    } 
    
    public function mealMenu() {
        return $this->belongsTo(MealMenu::class, 'meal_menu_id');
    }
  
    /**
     * @param $item
     *
     * @return void
     */
    function setFields($item): void
    {
        $menu = Menu::latest()->first();
        $item->menu_id = $menu->id;
        $item->data = Str::snake(Str::ascii($item->name, 'en'));
        $item->group = ESTIMATE_MENU_ITEM;
    }

}`

**Reply by phpfour** [reply]
Hi @amigolj, can you please share the output of `php artisan backpack:version` here so that we can reproduce using the exact version you're using?

**Reply by amigolj** [reply]
@phpfour   
![image](https://github.com/Laravel-Backpack/community-forum/assets/10392966/0079f414-c5ca-4fa7-98d4-2c1055e9609d)

**Comment by jcastroa87** [comment]
Hi @amigolj if not rule is apply on submit, did you check you have a call for "setValidation" of the request file at the beginning of the method?

```
CRUD::setValidation(StoreRequest::class);
```

Cheers.

**Comment by jcastroa87** [comment]
Due to not activity i will close this issue, but please feel free to re-open or create a new one if needed.

Cheers.
